en/developer_guide/content_add_field_listener.rst

```diff
@@ -0,0 +1,135 @@
+
+In some case, the field can store a collection which needs to be initialized by a listener to set all the datas.
+
+The main issues is that all the contentAttribute are stored as a raw field in mongoDB. By doing this, the data you will
+save will be typed but the data you get back will only be an array.
+
+Here is an example of how you can take profit of the Content architecture to store some TranslatedValue field
+inside.
+
+Usage
+-----
+
+Prerequisites
+~~~~~~~~~~~~~
+
+In the Open Orchestra model abstraction, the Translated Value are stored as a collection of ``TranslatedValueInterface``
+implementation. Each object will store the language code and the translated value.
+
+When you use it in a simple form, you will need to add a listener on the form :
+
+.. code-block:: php
+
+    $builder->addEventListener(FormEvents::PRE_SET_DATA, array($this->translateValueInitializerListener, 'preSetData'));
+
+Then you can simply add the form type :
+
+.. code-block:: php
+
+    $builder->add('descriptions', 'translated_value_collection');
+
+The entity should also implement the ``TranslatedValueContainerInterface``.
+
+Issue description
+~~~~~~~~~~~~~~~~~
+
+As you use a dynamic Field Type to describe your content, the ``Content`` document cannot implement the
+``TranslatedValueContainerInterface`` to give the name of the translated fields.
+
+Moreover, the listener has to be configured in the root form. As the configuration is fully dynamic, you
+cannot add it.
+
+Solution
+~~~~~~~~
+
+A simple solution is to create a specific form type, add an initializer, and take care of the data transformation.
+
+First create the specific form type :
+
+.. code-block:: php
+
+    class TranslatedValueContainerAttributeType extends AbstractType
+    {
+        public function getName()
+        {
+            return 'translated_value_container_attribute';
+        }
+    }
+
+
+Then inject the default value unitializer :
+
+.. code-block:: php
+
+    protected $translatedValueDefaultValueInitializer;
+
+    /**
+     * @param TranslatedValueDefaultValueInitializer $translatedValueDefaultValueInitializer
+     */
+    public function __construct(TranslatedValueDefaultValueInitializer $translatedValueDefaultValueInitializer)
+    {
+        $this->translatedValueDefaultValueInitializer = $translatedValueDefaultValueInitializer;
+    }
+
+Finally, create the listener to initialize the data :
+
+.. code-block:: php
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $translatedValueDefaultValueInitializer = $this->translatedValueDefaultValueInitializer;
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function(FormEvent $event) use ($translatedValueDefaultValueInitializer) {
+            $data = $event->getData();
+            if (is_null($data)) {
+                $values = new ArrayCollection();
+                $translatedValueDefaultValueInitializer->generate($values);
```

Comment:
Sorry, this documentation is prepared for the 0.2.3 version. In your current version, the class TranslatedValueDefaultValueInitializer does not exist.